### PR TITLE
docs(extra-files) add howto

### PR DESCRIPTION
**Pull request by sedlund**
Add a howto for `--extra-files`

#283

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -36,8 +36,8 @@ Options:
 * --stop-after-disko
   exit after disko formatting, you can then proceed to install manually or some other way
 * --extra-files <path>
-  path to a directory to copy into the root of the new nixos installation.
-  Copied files will be owned by root.
+  contents of local <path> are recursively copied and overwrites root (/) of the new NixOS installation.
```
Comment (phaer) [inline docs/cli.md:39]:
```suggestion
  contents of local <path> are recursively copied to the root (/) of the new NixOS installation. Existing files are overwritten
```

File: docs/howtos/extra-files.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+### Ownership
+
+The new system may have differing UNIX user and group id's for users created
+during installation.
+
+When the files are extracted on the remote the copied data will be owned by
+root.
+
+### Soft links
```
Comment (phaer) [inline docs/howtos/extra-files.md:73]: Nit: I think the term "symlinks" / "symbolic links" has won and woudl use that here

File: docs/howtos/extra-files.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+You want `/etc/ssh/ssh_host_*` and `/persist` from the local system on the
+target.  The `<path>` contents will look like this:
+
+```console
+$ cd /tmp
+$ root=$(mktemp -d)
+$ sudo cp --verbose --archive --link --parents /etc/ssh/ssh_host_* ${root}
+$ cp --verbose --archive --link /persist ${root} # --parents not needed for a full path
+$ lsd --tree --long ${root}
```
Comment (phaer) [inline docs/howtos/extra-files.md:30]: Maybe remove the lsd call and just post the output as an example layout in another source block? Using non-standard tools in a howto without explaining/installing them could be confusing?

File: docs/howtos/extra-files.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+```sh
+cd /tmp
+root=$(mktemp -d)
+cp --verbose --archive --link --parents /etc/ssh/ssh_host* ${root}
+```
+
+`--parents` will create the directory structure of the source at the
+destination.
+
+### Copy on Write (COW) file systems
```
Comment (phaer) [inline docs/howtos/extra-files.md:104]: Maybe I am missing something, but this is about space efficiency on the host you are installing from? In a context where the files often don't exceed a few kB?

Love the effort and examples, thank you! But would maybe drop this COW section completely

Comment (sedlund) [reply docs/howtos/extra-files.md:104]: wrt space, I was thinking about what would be in an impermanence setup like listed here:

https://github.com/nix-community/impermanence/blob/c6323585fa0035d780e3d8906eb1b24b65d19a48/README.org?plain=1#L398-L405

but the same case is covered by using hard links, so removed the COW section for brevity

**Comment by sedlund** [comment]
Appreciate the feedback!  Let me know if everything is addressed and I will squash.

**Comment by phaer** [comment]
Thank you!

**Pull request merged**